src/utils/logger.ts

```diff
@@ -0,0 +1,55 @@
+import { getFS } from "./window";
+
+export enum LogEventTypes {
+  DATALAYER_NOT_FOUND = 'Data Layer Not Found',
+  EMIT_FAILURE = 'Emit Failure',
+  OPERATOR_FAILURE = 'Operator Failure',
+  LISTENER_FAILURE = 'Listener Failure',
+  RULE_FAILURE = 'Rule Failure',
+  INCORRECT_ARGUMENTS = 'Incorrect Arguments',
+  UNKNOWN_INPUT = 'Unknown Input',
+}
```

Comment:
Docs TODO: The conditions that trigger these log event types should be enumerated.